### ISSUE TITLE
复现结果

### ISSUE THREAD

**Issue by Xxx-Nnn** [codgodtao/TMDiff#2]
您好，根据readme文档修改并训练模型，测试结果中包含很多零值，测试时在计算SAM时给出警告，请问您遇到过这样的问题吗

测试结果：
![Image](https://github.com/user-attachments/assets/4ed2dee7-cdbf-4914-a5b7-78a6994e9c8f)

警告信息：
/TMDiff-master/core/metrics.py:107: RuntimeWarning: invalid value encountered in divide
  res = np.arccos(dot_sum / norm_pred / norm_true)

**Comment by codgodtao** [comment]
抱歉，由于源代码的推理函数使用了
 x_recon = self.predict_start_from_noise(  # 根据预测出来的噪声得到x0
                x, t=t, noise=self.denoise_fn(x, time_in, x_in['PAN'], x_in['MS'], prompt))

这与训练函数中采用的x_recon = self.denoise_fn(x_noisy, time_in, x_in['PAN'], x_in['MS'], prompt)并不一致，这是代码归档阶段进行的消融实验导致的

SAM函数比较敏感，对于明显错误的数据会报错；


1. 如果您选择该训练函数，建议使用以下代码替换推理函数
@torch.no_grad()
    def super_resolution(self, x_in, continous, prompt, guidance):
        return self.sample_by_dpmsolver(x_in, prompt)
或者使用p_mean_variance_xo替换p_mean_variance函数

2. 如果您继续选择预测噪声的方式，则应该修改训练函数如下：
x_recon = self.denoise_fn(x_noisy, time_in, x_in['PAN'], x_in['MS'], prompt)
        loss = self.loss_func(noise, x_recon)
        return loss
